docker/Dockerfile

```diff
@@ -62,6 +62,21 @@ RUN apt-get update && \
     rm -rf /var/lib/apt/lists/*
 RUN /usr/local/bin/recursive_vcs_import.py src src/upstream
 
+# remove blacklisted packages from workspace
+ARG BLACKLISTED_PACKAGES_FILE="docker/blacklisted-packages.txt"
+ARG ENABLE_RECURSIVE_BLACKLISTED_PACKAGES="false"
+RUN echo "colcon list -p --base-paths src/ --packages-select \\" >> $WORKSPACE/.remove-packages.sh && \
+    if [[ $ENABLE_RECURSIVE_BLACKLISTED_PACKAGES == 'true' ]]; then \
+        find . -type f -name $(basename ${BLACKLISTED_PACKAGES_FILE}) -exec sed '$a\' {} \; | awk '{print "  " $0 " \\"}' >> $WORKSPACE/.remove-packages.sh ; \
+    elif [[ -f src/target/${BLACKLISTED_PACKAGES_FILE} ]]; then \
+        cat src/target/${BLACKLISTED_PACKAGES_FILE} | awk '{print "  " $0 " \\"}' >> $WORKSPACE/.remove-packages.sh ; \
+    fi && \
+    echo ";" >> $WORKSPACE/.remove-packages.sh && \
+    chmod +x $WORKSPACE/.remove-packages.sh && \
+    $WORKSPACE/.remove-packages.sh | xargs rm -rf
```

Comment:
Instead of removing the packages, `CATKIN_IGNORE` should also work if `COLCON_IGNORE` does not ([source](https://github.com/ros-infrastructure/rosdep/issues/786)). We can discuss whether we want to remove the packages or just ignore them.

Reply:
Discussion:
- we do not want to install the dependencies of these packages, nor have the src code in the dev image
- therefore: delete them completely